### PR TITLE
Use the official crystal docker image

### PR DESCRIPTION
**Pull request by jwaldrip**

### CODE REVIEW

**Comment by drujensen** [comment]
I agree, we should build off of the official one.

The latest version of Crystal is 0.23.1 and the official docker image has not been updated yet.

Also, we need 3 libraries not included in the official image: libsqlite3-dev libpq-dev libmysqlclient-dev.  

We will need a custom docker image for Amber since these libraries are not included.

File: src/templates/app/Dockerfile.ecr [hunk 1]
```diff
@@ -1,15 +1,16 @@
...
-RUN crystal deps
+# Build the app
+WORKDIR /app
+ADD shard.yml shard.lock /app
+RUN shards install
+RUN shards build
 
-CMD ["crystal", "spec"]
+# Set config
+CMD bin/app
```
Comment (drujensen) [inline src/templates/app/Dockerfile.ecr]: What is `bin/app`?

File: src/templates/app/Dockerfile.ecr [hunk 2]
```diff
@@ -1,15 +1,16 @@
...
+RUN cd /usr/local/share/amber_cmd-$AMBER_VERSION && shards install make
 RUN ln -sf /usr/local/share/amber_cmd-$AMBER_VERSION/bin/amber /usr/local/bin/amber
 
-WORKDIR /app/user
-
-ADD . /app/user
-
-RUN crystal deps
+# Build the app
+WORKDIR /app
```
Comment (drujensen) [inline src/templates/app/Dockerfile.ecr]: :+1:

**Comment by drujensen** [comment]
@jwaldrip We also use `netcat` to determine if the db is running.  Can you add the following to the `DockerFile.ecr`?

```
RUN apt-get install -y --no-install-recommends netcat libsqlite3-dev libpq-dev libmysqlclient-dev
```

Also, change `/app/user` to `/app` in the `docker-compose.yml` file.

**Comment by jwaldrip** [comment]
How about creating a amber base image for amber based off of the crystal
image?

File: src/templates/app/Dockerfile.ecr [hunk 3]
```diff
@@ -1,15 +1,16 @@
-FROM drujensen/crystal:0.23.0
+FROM crystallang/crystal:0.23.0
```
Comment (eliasjpr) [inline src/templates/app/Dockerfile.ecr]: Shouldn't this be version `0.23.1`?

**Comment by elorest** [comment]
I'm for using the default crystal on but it's usually behind and I'm not sure who has write access to it.

**Comment by elorest** [comment]
@drujensen As far as those three libraries we can just tell our docker file to install them.

**Comment by jwaldrip** [comment]
I will make the updates today.

**Comment by jwaldrip** [comment]
closing in favor of #71

**Pull request closed without merging**